docs/source/guides/monitoring.rst

```diff
@@ -1,12 +1,188 @@
+==========================
 Monitoring with Prometheus
 ==========================
 
+Monitoring stacks usually consist of a metrics collector, a time-series database to store metrics
+and a visualization layer. A popular stack is `Prometheus <https://prometheus.io/>`_ with `Grafana <https://grafana.com/>`_
+used as the visualization layer to create rich dashboards. An architecture of Prometheus and its ecosystem is shown below:
+
+.. image:: ../_static/img/prom-architecture.png
+
+
+BentoML API server comes with Prometheus support out of the box. When launching an API model server with BentoML,
+whether it is running dev server locally or deployed with docker in the cloud, a ``/metrics`` endpoint will always
+be available for exposing prometheus metrics. This guide will introduce how you use Prometheus and Grafana to monitor
+your BentoService.
+
+.. note::
+    Currently custom metrics is not yet supported in current version of BentoML. We are working towards providing
```

Comment:
I think users can already use the Prometheus client to expose custom metrics in their BentoService code?

Reply:
I will just remove this note then.

Reply:
@parano are there any examples, or can you provide any, for how this would be done?

Reply:
We could open a different ticket regarding custom metrics support. As of right now, I'm not aware of any way for end users to achieve this. 

https://github.com/bentoml/BentoML/blob/ffacfc50286488bd22d4995fd720fe54bc15abc6/bentoml/server/instruments.py#L15 This is the middleware that handles server metrics.